src/Controller/Component/FlashComponent.php

```diff
@@ -83,20 +93,159 @@ public function set($message, array $options = [])
             $message = $message->getMessage();
         }
 
-        list($plugin, $element) = pluginSplit($options['element']);
+        $options['element'] = $this->_getElement($options['element']);
 
-        if ($plugin) {
-            $options['element'] = $plugin . '.Flash/' . $element;
-        } else {
-            $options['element'] = 'Flash/' . $element;
+        $sessionKey = 'Flash.' . $options['key'];
+        $messages = [];
+        $index = 0;
+
+        if ($options['reset'] === false) {
+            $messages = $this->_session->read($sessionKey);
+        }
+
+        if (!empty($messages)) {
+            end($messages);
+            $index = key($messages);
+            reset($messages);
+            $index++;
```

Comment:
I did this at first but that brought problems if we can delete specific messages. I we use count, this :

```php
$this->Flash->success('0');
$this->Flash->success('1');
$this->Flash->success('2');
$this->Flash->delete(1);
$this->Flash->success('3');
```

will output 0 and 3 because, after deletion, the count is 2 and we already have a message indexed at 2. Which is why I had to use this.
But since we are killing ``delete()`` and ``clean()``, ``count()`` can be back in the game.